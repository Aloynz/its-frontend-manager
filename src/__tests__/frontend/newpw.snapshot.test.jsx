import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import ResetPasswordView from '@/app/(auth)/new-password/page'
import { ReadonlyURLSearchParams } from 'next/navigation';

// mock useSearchParams
jest.mock('next/navigation', () => ({
    useSearchParams: () => new URLSearchParams({ 'token': 'mockToken' }),
  }));

it("renders new password page unchanged", async () => {
    const result = await ResetPasswordView()
    const { container } = render(result);
    expect(container).toMatchSnapshot();
  });
