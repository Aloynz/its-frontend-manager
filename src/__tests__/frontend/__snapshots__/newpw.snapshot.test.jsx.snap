// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders new password page unchanged 1`] = `
<div>
  <div
    class="flex justify-center items-center h-screen"
  >
    <div
      class="max-w-md p-8 bg-white rounded-lg shadow-lg"
    >
      <h2
        class="text-2xl font-semibold mb-7 text-center"
      >
        Reset your password
      </h2>
      <form>
        <div
          class="space-y-2"
        >
          <label
            class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
            for=":r0:-form-item"
          >
            New password
          </label>
          <div
            aria-describedby=":r0:-form-item-description"
            aria-invalid="false"
            id=":r0:-form-item"
            style="display: flex; align-items: center; margin-bottom: 10px; margin-top: 10px;"
          >
            <svg
              class="lucide lucide-lock "
              fill="none"
              height="19"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              style="margin-right: 8px;"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="11"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="11"
              />
              <path
                d="M7 11V7a5 5 0 0 1 10 0v4"
              />
            </svg>
            <input
              class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
              name="password"
              placeholder="e.g. password123"
              type="password"
            />
          </div>
        </div>
        <div
          style="text-align: center; padding-bottom: 15px; padding-top: 10px;"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-10 px-4 py-2"
            style="width: 200px; border-radius: 10px;"
            type="submit"
          >
            Change password
          </button>
        </div>
      </form>
      <hr
        style="border-color: gray;"
      />
      <div
        class="mt-4 flex items-center justify-center"
      >
        <p
          class="ml-2"
        >
          Have an account? 
        </p>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 text-primary underline-offset-4 hover:underline h-10 px-4 py-2"
        >
          <a
            href="/signin"
          >
            Login now
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
